responsiveness upgrade

deadroxy opened the pull request:
This initial commit introduces the flexible box model via the Flexbox Ultralight library. It makes minimal changes necessary to render the about page body (but not the header or footer) responsive, without (hopefully) breaking the rest of the site. This code is not ready for production, issuing the pull request so other can see the work in progress.

jcn commented:
@deadroxy FYI #99 has been merged into master.

deadroxy commented:
Just pushed all my latest code which...
- Makes the hamburger menu sitewide, responsive and a bit prettier
- Makes the footer responsive
- Ensures that the about page is properly responsive
- Makes the FAQ responsive
- Makes the application form responsive
- Makes all the logged in forms responsive (as a bonus)
- Makes the user/session forms prettier but not entirely responsive (because I can't actually change non-existent views)
- Merges with the upstream master to incorporate Rails and Owl upgrades
- Refactors the application layout so that responsiveness stops breaking the image carousels/matrix
- Makes preliminary progress on chapter page responsiveness

jcn commented:
@deadroxy I merged in #359 which should help clean up some of that one-off clearance styling.

deadroxy commented:
@Jesse FYI just added a bunch of commits working through responsiveness on the chapter show page. Here's a summary:
- DRY up mobile responsive base variable sizing and create clear definitions.
- Ensure margin on top of admin panel disappears when hamburger menus appear.
- Make owl carousel on chapter show pages responsive.
- Prevent vertical scrolling of chapter page on mobile devices when touch scrolling inside Owl carousel.
- Make chapter show pages fully responsive.
- Begin work on project show page by ensuring buttons/links at the top are properly sized.

I also started working through the project show pages, but encountered some initial difficulty because the carousel there is implemented differently from the new Owl stuff on the chapter show pages. I assume we want that gallery to scale smoothly, but so far that's proving tricky. Part of that is because I can't upload images when developing locally so it would be good to figure out if there's a way to do that for development purposes.

deadroxy commented:
@jcn added my "final" chunk of commits which represent a first pass at "completion" on this project. In the latest commits you will find:

- Public project show pages are now fully responsive, including the image galleries
- Homepage is now fully responsive, including the image matrix

I worked out a way of dealing with development on the project show pages even though I couldn't upload images. Obviously, both the project galleries and image matrix on the homepage are affected by this dev issue because I cannot actually test with live images. Nonetheless, I've delivered the most unbroken version of this code I can without a way to test on actual images 😃

jcn commented:
This is looking great. There are a few more things I need to test with better data, but one thing is I think the Chapters dropdown would look better in some sort of grid - right now at full-width everything is flowing kind of crazy.

This is what it's doing now:
![Screen Shot 2019-04-14 at 10 06 25 PM](https://user-images.githubusercontent.com/2909/56103580-97544880-5f01-11e9-9093-0d5607816991.png)

If we can't get a grid to work, maybe we can just do each country as its own section and have each chapter listing be a fixed width (33%, 20%, whatever looks reasonable). Something like this?

![Screen Shot 2019-04-14 at 9 15 14 PM](https://user-images.githubusercontent.com/2909/56103529-5d834200-5f01-11e9-9df4-78f54da6527b.png)

We can discuss further before implementation to talk through pitfalls if you'd like.

deadroxy commented:
@jcn Just pushed a new commit with my proposal for an improved chapter dropdown flow. I took a look at all the live data (for string lengths) and determined a 5 column grid style works best at maximum width, and worked back from there. This approach maximizes grid/readability and still conserves whitespace. Let me know if it looks good to you!

jcn commented on app/views/acceptances/new.html.erb:
```diff
@@ -1,6 +1,13 @@
-<% end %>
+<div class="flexbox vertical center">
+  <section class="invitation-form">
+    <h1>Accept the trustee invitation</h1>
```
We shouldn't have any actual strings of text in the views - this string should be moved to the `config/locales/en.yml` file under the `acceptances/new/title` key and then we can get it translated into other languages in the future. Also when it's moved let's make this "Accept your trustee invitation"

deadroxy replied:
This code was added because this form remained unstyled during the clearance upgrade. It adopts the same pattern found in `app/views/invitations/new.html.erb`. That form also has the actual string of text in the view. So, both ought to be fixed then.

jcn replied:
OK, we'll take care of this in another commit.

jcn commented on app/views/acceptances/new.html.erb:
```diff
@@ -1,6 +1,13 @@
+        <%= form.input :last_name, :required => true %>
+      </div>
+      <%= form.input :password, :required => true %>
+      <%= submit_tag "Accept" %>
```
Same as above, this should be moved into the `config/locales/en.yml` under the `acceptances/new/accept` key file

deadroxy replied:
Same as my comment above, the submit tag in the `app/views/invitations/new.html.erb` form which this mirrors is also hardcoded English text, and should similarly be changed.

jcn commented:
Oh, another thing I noticed - for the forms where the fields are side-by-side when they are full-sized, when they collapse to a single column, the form fields do not stretch to full-width. Any insights into why this might be? 

![Screen Shot 2019-04-16 at 1 31 06 PM](https://user-images.githubusercontent.com/2909/56231217-e9a18080-604b-11e9-94fd-95d0948e9e88.png)

![Screen Shot 2019-04-16 at 1 29 29 PM](https://user-images.githubusercontent.com/2909/56231198-ddb5be80-604b-11e9-93f9-0d515124c5c4.png)

deadroxy commented:
Half-width (actually 220px) form fields are peppered throughout a variety of forms on the site. The reason they don't go full width when the forms are squeezed in size is because they have a fixed width. Of course, it is possible to go through all the forms and create a responsive point to make these fields grow once they reach the single column state. It doesn't come automagically so it would need explicit doing is all.

deadroxy commented:
@jcn Just pushed some code to fix the responsiveness of the owl dots on chapter show pages. In their native responsive state the dots don't actually confine themselves to a single line as you thought. The demos you have seen on owl make use of the dotsEach setting as applied to the responsive break points. The code I added *should* work to confine the dots to a single line, but it depends on the underlying data. So let's push this to preview and see how it looks.

deadroxy commented:
@jcn Scratch that. Just did some testing and I want to try a different solution. Hold please.

deadroxy commented:
@jcn Okay, pushed a better solution!

deadroxy commented:
@jcn Alright. I think that's everything you asked for. May all your wishes come true, etc. In this final series of ultra bonus commits I have:
- Implemented responsiveness for Owl dots on chapter show pages
- Created a responsive paradigm for half-width form fields
- Applied this paradigm to project and invitation forms
- Adjusted margin on top of admin panel when burgers are engaged
- Ensured chapter show page is full-width singe-column
- Improved hover on image matrix at small sizes
- Improved view of project list on admin pages at small sizes
- Made admin project filter menu responsive

After 46 hours deep in the codes, Awesome is now... totally (fingers crossed) responsive!

deadroxy commented:
PS: Found another small bug (on the sign-in/password forms) from the clearance roll-over and fixed it.

deadroxy commented:
PPS: Adjusted some font sizes on the admin pages. That's it. I'll stop tweaking!

jcn commented:
@deadroxy I'm having folks play around with the site now to figure out whether we missed anything. 

One thing that's come up (that we've discussed in the past, but which we'd like to revisit) is the size of the images on the project page at different breakpoints. I'd like to get a sense of the scope of a project to update the page so images on the project page could always flow to a maximum width on the page, rather than just resizing to the 3 sizes they currently do. 

![Screen Shot 2019-04-18 at 12 04 27 AM](https://user-images.githubusercontent.com/2909/56336250-9c1d3480-616d-11e9-874d-16ea1a8c3f8e.png)

vs

![Screen Shot 2019-04-17 at 11 58 38 PM](https://user-images.githubusercontent.com/2909/56336167-48125000-616d-11e9-8cfc-b03114b74b0f.png)

deadroxy commented:
I've merged in your PR @jcn (with a tiny change) and brought this branch up to date with master (fixing a bunch of conflicts in the process). Do you want to merge this PR into master and deploy that code to production so we can be done with this branch? I'm going to start on the project page image tiling in a new branch if this is all set!

jcn commented:
@deadroxy Were you going to make the adjustments to the homepage spacing in this PR or did you want to do it after we merge into master?

deadroxy commented:
@jcn we identified four additional fixes (listed below) in our last meeting about the project galleries. I was planning to do these alongside the project gallery work so we could close out and move on from this PR.

- Fix Owl carousel when displaying less than three images
- Adjust whitespace on the homepage
- Adjust font size on user lists
- Make admin chapter name dropdown bounded on non-touch devices

deadroxy commented:
Per your request, @jcn, I've added the 4 fixes mentioned in the previous comment in this last series of commits to this branch/PR. For reference those fixes are:
- Ensures Owl carousel looks nice when displaying less than three images
- Ensures all text on user lists is reasonably sizes
- Ensures chapter selection list dropdown in the admin panel is bounded when opened on non-touch devices
- Adjusts whitespace on homepage to be more pleasant

jcn commented:
> Ensures all text on user lists is reasonably sizes

The font sizes look good here, but there appears to be a stray padding issue (Firefox, Mac).
![Screen Shot 2019-04-30 at 12 40 54 AM](https://user-images.githubusercontent.com/2909/56941228-2e63f780-6ae1-11e9-906d-8cfc8603ce61.png)

deadroxy commented:
@jcn, just committed five small fixes. This branch is super ready for deployment now. Fixes include:
- Plugging in a mouse no longer messes up user lists
- Font size adjusted for link to public project page in admin dashboard view
- Font size adjusted for 'Back to Chapter' link on admin view of project page
- Font size adjusted for pagination
- Font size adjusted on finalists lists

PS: The pagination definitely is Flickr style ;]
<img width="463" alt="Screen Shot 2019-04-30 at 2 54 39 AM" src="https://user-images.githubusercontent.com/126442/56946229-84439a00-6af7-11e9-88ee-45e6d507657e.png">

deadroxy commented:
Just noticing a failure from Travis but I've never seen this sort of error before. Let me know if you have any insight @jcn.

jcn commented:
> Just noticing a failure from Travis

This appears to be a change that they've made to how we run headless tests on their system. A commit has been made to master that fixes this (#373).

deadroxy commented:
@jcn that seemed to do the trick!

deadroxy commented:
@jcn a few lines of smart css have been added, and pagination is now streamlined for smaller window sizes.

jcn commented:
This looks great. Going to merge this in now!
